pypesto/store/save_to_hdf5.py

```diff
@@ -182,6 +185,11 @@ def write(self, result: Result, overwrite: bool = False):
         """
         Write HDF5 sampling file from pyPESTO result object.
         """
+        # if there is no sample available, raise a warning and return
+        if(isinstance(result.sample_result, SampleResult)):
```

Comment:
Maybe add a comment about `SampleResult` being some dummy type that indicates no sampling has already been done (actual sampling is indicated by `McmcPtResult`)

Reply:
```suggestion
        # SampleResult is only a dummy class created by the Result class
        if(isinstance(result.sample_result, SampleResult)):
```